flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommitRequestImpl.java

```diff
@@ -32,16 +33,27 @@ public class CommitRequestImpl<CommT> implements Committer.CommitRequest<CommT>
     private CommT committable;
     private int numRetries;
     private CommitRequestState state;
+    private SinkCommitterMetricGroup metricGroup;
 
-    protected CommitRequestImpl(CommT committable) {
+    protected CommitRequestImpl(CommT committable, SinkCommitterMetricGroup metricGroup) {
         this.committable = committable;
+        this.metricGroup = metricGroup;
         state = CommitRequestState.RECEIVED;
+
+        // Currently only the SubtaskCommittableManager uses this constructor to create a new
+        // CommitRequestImpl, so we can increment the metrics here
+        metricGroup.getNumCommittablesTotalCounter().inc();
```

Comment:
this feels a bit hacky, as your comment already hints. My main issue with this is that this happening in the constructor, if we only look at this class locally it's hard to tell if we're incrementing it correctly. For example, a deserializer for `CommitRequestImpl`s can totally call this constructor and unintentionally increment the metric.

I see that you've already wired in the `SinkCommitterMetricGroup` to the `SubtaskCommittableManager`. Can we just increment the total # of committables there in the `add` method then?

Reply:
Moved.